### ISSUE TITLE
feat: improve docker setup

### ISSUE THREAD

**Issue by Nsttt** [t3-oss/create-t3-app#603]
### Is your feature request related to a problem? Please describe.

Currently in the docs it states that I should remove the `import { env } from "./src/env/server.mjs";` when deploying with Docker. 

I believe this work as a placeholder and there should be some sort of guide / system that allows users an easy-going experience when deploying with Docker.

Maybe some pre-defined files in the templates would be good, Or just docs are enough.

### Describe the solution you'd like to see

An easy-going setup with docker.

Allow for different configs in `dev` or `prod` and guide the user to deploy the app.

### Desribe alternate solutions

Probably split the config files into 2 one for prod one for development?

### Additional information

I'm looking into this at the time of writting this, so any progress I'll update it here.

Any opinions and feedback is much appreciated.

**Comment by c-ehrlich** [comment]
I keep forgetting to write the docs on how to set up a docker dev setup with HMR etc. But obviously that alone isn't good enough - the real goal would be to be able to develop locally and deploy to docker, without having to use different branches or some other hack.

If this requires a complex setup (ie multiple config files etc) I think we should not include it in the template, but give instructions in the docs.

Curious to see what you come up with :)

**Comment by Nsttt** [comment]
I've made some progress on this over the week. 

So far I've managed to dinamically import the enviroment into the `next.config.mjs`. To do this, in order to use top-level await I have changed the target in tsconfig to `es2017` **still unsure about this and I would like some feedback.**

Following, there should also be a few more changes into the docs pages explaining.
- How to setup and prod enviroment and everything to keep in mind on settings.
- How to setup and dev enviroment.
- Probably a more detailed view of working with enviromental variables ?

Anything else you feel is needed please add below. I'll make a PR with the changes soon. 😄

**Comment by c-ehrlich** [comment]
What does targeting es2017 mean in this context? It's at 96% on caniuse, so would it mean incompatibility with 4% of web users?

**Comment by Nsttt** [comment]
To your first question, I believe 

To your second question, to my knowledge, moving away from `es5` loses compatibility with IE11, which I don't find much of an issue and not much else. Change, allows us to use features from es6, so in the `next.config.mjs`, we can add the following snippet.

```ts
process.env.NODE_ENV === "development"
  ? await import("./src/env/server.mjs")
  : null;
```

And so, get a conditionally imported `env`.

**Comment by juliusmarminge** [comment]
Resolved in #646
